scripts/actions/mobskills/everyones_grudge.lua

```diff
@@ -19,12 +19,29 @@ mobskillObject.onMobWeaponSkill = function(target, mob, skill)
     local power   = 5
     local realDmg = power * target:getCharVar('EVERYONES_GRUDGE_KILLS') -- Damage is 5 times the amount you have killed
 
-    if target:getID() > 100000 then
+    -- TODO: Verify if this is accurate
+    if target:isPet() then
         realDmg = power * math.random(30, 100)
-    elseif mob:isNM() then
+    end
+
+    if mob:isNM() then
         realDmg = realDmg * 10 -- sets the multiplier to 50 for NM's
     end
 
+    -- Uggalepih Necklace mitigation
+    -- While worn, consumes all TP to mitigate damage at flat breakpoints
+    -- 1500 TP = 50% reduction, 3000 TP = 100% reduction
+    if target:getEquipID(xi.slot.NECK) == xi.item.UGGALEPIH_NECKLACE then
+        local targetTP = target:getTP()
+        if targetTP == 3000 then
+            realDmg = 0
+        elseif targetTP >= 1500 then
+            realDmg = realDmg * 0.5
+        end
+
+        target:setTP(0)
+    end
```

Comment:
```lua
    if target:getEquipID(xi.slot.NECK) == xi.item.UGGALEPIH_NECKLACE then
        local tpFactor = 1 - 0.5 * math.floor(target:getTP() / 1500)
        realDmg        = math.floor(realDmg * tpFactor)

        target:setTP(0)
    end
```

Reply:
Got that adjusted for both moves, also added in a check before the slot to check if target is PC in case that might be an issue checking that on non player entities.